ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -32,6 +32,37 @@ message Completion {
   // Optional
   string transaction_id = 3;
 
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  string application_id = 6;
+
+  // The set of parties on whose behalf the commands were executed.
+  // Contains the union of `party` and `act_as` from ``commands.proto``.
+  // The order of the parties need not be the same as in the submission.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  repeated string act_as = 7;
+
+  // The submission ID this completion refers to, as described in ``commands.proto``.
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  // TODO(v2): Do we want to enforce the format of a UUID here?
+  string submission_id = 4;
+
+  // The submission rank of the submission this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  // TODO(v2): Can be made required once all Daml ledgers return the application ID
+  string submission_rank = 5;
```

Comment:
@andreaslochbihler-da : can you remind me again why this information is required?

Reply:
Best practice of including all information that is relevant for expressing properties. For example:
* If you want to audit the completion stream of a participant, then you can check the rank guarantee without having to keep a lookup table from submission UUIDs to ranks.
* Some ledgers cannot enforce that the submission UUID is indeed unique because the application chooses the UUID. In those cases, one could not even build such a sensible lookup table.